vhost-user-backend/src/event_loop.rs

```diff
@@ -135,13 +185,31 @@ where
     }
 
     pub(crate) fn register_event(&self, fd: RawFd, ev_type: EventSet, data: u64) -> Result<()> {
-        self.epoll
-            .ctl(ControlOperation::Add, fd, EpollEvent::new(ev_type, data))
+        let mut fd_set = self.fd_set.lock().unwrap();
+        if fd_set.contains(&fd) {
+            return Err(io::Error::from_raw_os_error(libc::EEXIST));
+        }
+        self.registry
+            .register(
+                &mut SourceFd(&fd),
+                Token(data as usize),
```

Comment:
since `data: u64`  and `pub struct Token(pub usize);` are we sure this cannot be truncated?

Reply:
nit: maybe we can move these changes in a separate commit (same PR, before adding `mio`)